app.py

```diff
@@ -8,157 +10,216 @@
+                    request.method,
+                    request.path,
+                )
+                + s
+            )
         except:
-            return '[%s] [SYS] '%self.formatTime(record) + s
+            return "[%s] [SYS] " % self.formatTime(record) + s
+
 
 if not app.debug:
-    mail_handler = SMTPHandler(app.config['SMTP_SERVER'],
-                                app.config['APP_EMAIL'],
-                                app.config['ADMIN_EMAIL'], 'LetterBomb ERROR')
+    mail_handler = SMTPHandler(
+        app.config["SMTP_SERVER"],
+        app.config["APP_EMAIL"],
+        app.config["ADMIN_EMAIL"],
+        "LetterBomb ERROR",
+    )
     mail_handler.setLevel(logging.ERROR)
     app.logger.addHandler(mail_handler)
 
-    handler = logging.FileHandler(os.path.join(app.root_path, 'log', 'info.log'))
+    handler = logging.FileHandler(os.path.join(app.root_path, "log", "info.log"))
     handler.setLevel(logging.INFO)
     handler.setFormatter(RequestFormatter())
     app.logger.addHandler(handler)
 
     app.logger.setLevel(logging.INFO)
-    app.logger.warning('Starting...')
+    app.logger.warning("Starting...")
+
 
 def region():
     try:
         country = gi.country(request.remote_addr).country.iso_code
         app.logger.info("GI: %s -> %s", request.remote_addr, country)
-        return COUNTRY_REGIONS.get(country, 'E')
+        return COUNTRY_REGIONS.get(country, "E")
     except:
         app.logger.exception("GeoIP exception")
-        return 'E'
+        return "E"
+
 
 def _index(error=None):
-    g.recaptcha_args = 'k=%s' % app.config['RECAPTCHA_PUBLICKEY']
-    rs = make_response(render_template('index.html', region=region(), error=error))
-    #rs.headers['Cache-Control'] = 'private, max-age=0, no-store, no-cache, must-revalidate'
-    #rs.headers['Etag'] = str(random.randrange(2**64))
-    rs.headers['Expires'] = 'Thu, 01 Dec 1983 20:00:00 GMT'    
+    g.recaptcha_args = "k=%s" % app.config["RECAPTCHA_PUBLICKEY"]
+    rs = make_response(render_template("index.html", region=region(), error=error))
+    # rs.headers['Cache-Control'] = 'private, max-age=0, no-store, no-cache, must-revalidate'
+    # rs.headers['Etag'] = str(random.randrange(2**64))
+    rs.headers["Expires"] = "Thu, 01 Dec 1983 20:00:00 GMT"
     return rs
 
 
-@app.route('/')
+@app.route("/")
 def index():
     return _index()
 
 
 def captcha_check():
     try:
-        oform = {
-            #"privatekey": app.config['RECAPTCHA_PRIVATEKEY'],
-            "secret": app.config['RECAPTCHA_PRIVATEKEY'],
-            "remoteip": request.remote_addr,
-            #"challenge": request.form.get('recaptcha_challenge_field',['']),
-            #"response": request.form.get('recaptcha_response_field',[''])
-            "response": request.form.get('g-recaptcha-response',[''])
+        payload = {
+            "response": request.form.get("g-recaptcha-response", [""]),
+            "secret": app.config["RECAPTCHA_PRIVATEKEY"],
```

Comment:
Is 'remoteip' not useful for the recaptcha service?

Reply:
its not used. as far as I've seen i get errors when i send it.

Reply:
It's valid, see [here](https://developers.google.com/recaptcha/docs/verify), and it's working fine in the python2 version. If it's not working for you then there is some other issue.

Reply:
I have added it to the parameters sent.